Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -2547,7 +2547,21 @@ void MediaPlayerPrivateGStreamer::configureParsebin(GstElement* parsebin)
             return tryAutoPlug;
         }), this);
 
-    // We need to ensure that the webkitthunderparser factory is preferred over other parsers.
+#if ENABLE(ENCRYPTED_MEDIA) && ENABLE(THUNDER)
+    // Inject the webkitthunderparser factory into parsebin's autoplug-factories signal.
+
+    // The legacy decodebin2 stack doesn't integrate well with parsebin, so prevent auto-plugging of
+    // the webkitthunderparser.
+    if (m_isLegacyPlaybin)
+        return;
```

Comment:
This can be removed I think, `configureParsebin()` isn't called in legacy playbin case, afaik.